ledger/src/blockstore_processor.rs

```diff
@@ -1051,6 +1054,23 @@ fn process_bank_0(
     bank0.freeze();
     blockstore.insert_bank_hash(bank0.slot(), bank0.hash(), false);
     cache_block_meta(bank0, cache_block_meta_sender);
+
+    if let Some(snapshot_config) = snapshot_config {
```

Comment:
I could also add another check `&& snapshot_config.full_snapshot_archive_interval_slots != Slot::MAX` to further only take a snapshot if snapshots are fully enabled.